specification/dashboard/resource-manager/Microsoft.Dashboard/preview/2023-10-01-preview/grafana.json

```diff
@@ -0,0 +1,2419 @@
+        "name": "ManagedPrivateEndpointConnectionStatus"
+      }
+    },
+    "ManagedPrivateEndpointUpdateParameters": {
+      "description": "The parameters for a PATCH request to a managed private endpoint.",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "description": "The new tags of the managed private endpoint.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "IntegrationFabric": {
+      "type": "object",
+      "description": "The integration fabric resource type.",
+      "x-ms-azure-resource": true,
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/IntegrationFabricProperties"
+        }
+      }
+    },
+    "IntegrationFabricListResponse": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IntegrationFabric"
+          }
+        },
+        "nextLink": {
+          "type": "string"
+        }
+      }
+    },
+    "IntegrationFabricProperties": {
+      "type": "object",
+      "properties": {
+        "provisioningState": {
+          "description": "Provisioning state of the resource.",
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "targetResourceId": {
+          "description": "The resource Id of the Azure resource being integrated with Azure Managed Grafana. E.g., an Azure Kubernetes Service cluster.",
+          "type": "string"
+        },
+        "dataSourceResourceId": {
+          "description": "The resource Id of the Azure resource which is used to configure Grafana data source. E.g., an Azure Monitor Workspace, an Azure Data Explorer cluster, etc.",
+          "type": "string"
+        },
+        "scenarios": {
+          "description": "A list of integration scenarios covered by this integration fabric",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "IntegrationFabricUpdateParameters": {
+      "description": "The parameters for a PATCH request to a Integration Fabric resource.",
+      "type": "object",
+      "properties": {
+        "tags": {
+          "description": "The new tags of the Integration Fabric resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "description": "The new properties of this Integration Fabric resource",
+          "$ref": "#/definitions/IntegrationFabricPropertiesUpdateParameters"
+        }
+      }
+    },
+    "IntegrationFabricPropertiesUpdateParameters": {
+      "type": "object",
+      "properties": {
+        "scenarios": {
```

Comment:
>scenarios [](http://example.com/codeflow?start=9&length=9)

are there any fixed set of scenarios possible ? so that this can be modeled as enum instead of a free string ?

Reply:
Was discussed previously https://github.com/Azure/azure-rest-api-specs-pr/pull/19253#discussion_r1794606289 and need to keep it as a string.